packages/dx-grid-core/src/plugins/grouping-state/reducers.ts

```diff
@@ -8,7 +8,7 @@ import {
 const applyColumnGrouping: PureReducer<Grouping[], ChangeGroupingPayload> = (
   grouping, { columnName, groupIndex },
 ) => {
-  const nextGrouping = grouping.slice();
+  const nextGrouping = slice(grouping);
```

Comment:
If none of the following conditions is true then the function result is falsely different - `grouping` is not changed actually but instance reference is changed.